packages/mobx-state-tree/src/core/mst-operations.ts

```diff
@@ -789,6 +789,41 @@ export function getMembers(target: IAnyStateTreeNode): IModelReflectionData {
     return reflected
 }
 
+export type CastedType<T> = T extends IStateTreeNode<infer STNC> ? STNC | T : T
```

Comment:
Maybe comment what `STNC` stands for?

Reply:
it just means State Tree Node Creation type, just renamed it to C